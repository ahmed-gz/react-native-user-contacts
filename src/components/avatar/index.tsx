import { StyleSheet, Text, View } from "react-native";

type Props = {
  name: string;
  variant?: "small" | "large";
};

export function Avatar({ name, variant = "large" }: Props) {
  const isLarge = variant === "large";
  const size = isLarge ? 70 : 30;

  return (
    <View
      style={[
        styles.avatar,
        {
          width: size,
          height: size,
          borderRadius: size,
        },
      ]}
    >
      <Text
        style={[
          styles.avatarName,
          {
            fontSize: isLarge ? 45 : 16,
          },
        ]}
      >
        {name[0].toUpperCase()}
      </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  avatar: {
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#ccc",
  },
  avatarName: {
    color: "#fff",
    fontWeight: "bold",
  },
});
