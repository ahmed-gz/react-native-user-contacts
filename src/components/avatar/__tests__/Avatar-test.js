import renderer from "react-test-renderer";

import { Avatar } from "@components/avatar";

it(`renders correctly`, () => {
  const tree = renderer.create(<Avatar name="Jon" />).toJSON();

  expect(tree).toMatchSnapshot();
});
