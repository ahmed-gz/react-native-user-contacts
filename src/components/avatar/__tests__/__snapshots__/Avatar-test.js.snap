// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#ccc",
        "justifyContent": "center",
      },
      {
        "borderRadius": 70,
        "height": 70,
        "width": 70,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
          "fontWeight": "bold",
        },
        {
          "fontSize": 45,
        },
      ]
    }
  >
    J
  </Text>
</View>
`;
